### PR TITLE
EMF fixes

### PR DESCRIPTION
**Pull request by enikao**
Include more files into emf-builtins artifact, so it can be recognized as Eclipse plugin.

Implemented workaround for #112.

Added more info to exceptions.

### CODE REVIEW

File: emf-builtins/META-INF/MANIFEST.MF [hunk 1]
```diff
@@ -0,0 +1,6 @@
+Manifest-Version: 1.0
+Bundle-ManifestVersion: 2
+Bundle-Name: emf-builtins
+Bundle-SymbolicName: io.lionweb.java.emf.builtins;singleton:=true
+Bundle-Version: 0.1.0.qualifier
```
Comment (ftomassetti) [inline emf-builtins/META-INF/MANIFEST.MF:5]: Is this the version of the Maven package? If so, we should consider generating this file

Comment (ftomassetti) [reply emf-builtins/META-INF/MANIFEST.MF:5]: Looking at the build.gradle file it seems generated, so I think we should not commit it, but adding the task generating this file as a dependency of the `setup` task

File: emf/src/main/java/io/lionweb/lioncore/java/emf/mapping/ConceptsToEClassesMapping.java [hunk 2]
```diff
@@ -121,6 +136,28 @@ public void registerMapping(Interface iface, EClass eClass) {
     interfacesToEClasses.put(iface, eClass);
   }
 
+  public void registerMapping(Language language, EPackage ePackage) {
+    for (LanguageEntity entity : language.getElements()) {
```
Comment (ftomassetti) [inline emf/src/main/java/io/lionweb/lioncore/java/emf/mapping/ConceptsToEClassesMapping.java:140]: Should we check if this mapping already exists and in that case just return?
Should we check for non-conflict mapping?

File: emf/src/test/resources/properties.ecore [hunk 3]
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <ecore:EPackage xmi:version="2.0" xmlns:xmi="http://www.omg.org/XMI" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-    xmlns:ecore="http://www.eclipse.org/emf/2002/Ecore" name="io.lionweb.Properties" nsURI="https://lionweb.io/io_lionweb_Properties"
+    xmlns:ecore="http://www.eclipse.org/emf/2002/Ecore" name="io_lionweb_Properties" nsURI="https://lionweb.io/io_lionweb_Properties"
```
Comment (ftomassetti) [inline emf/src/test/resources/properties.ecore:3]: Wouldn't be better to keep the dots? Or is there a reason to replace them?

**Comment by ftomassetti** [comment]
Thank you for the ton of improvements!

**Review by ftomassetti — APPROVED** [review]

File: emf/src/main/java/io/lionweb/lioncore/java/emf/EMFMetamodelExporter.java [hunk 4]
```diff
@@ -32,7 +32,7 @@ public Resource exportResource(List<Language> languages) {
   public EPackage exportLanguage(Language language) {
     EPackage ePackage = EcoreFactory.eINSTANCE.createEPackage();
 
-    ePackage.setName(language.getName());
+    ePackage.setName(language.getName().replaceAll("[^\\p{Alnum}_]", "_"));
```
[the remark this answers is not in the document]
Comment (enikao) [reply emf/src/main/java/io/lionweb/lioncore/java/emf/EMFMetamodelExporter.java:35]: Yes, dots lead to invalid generated Java. I investigated the code of the `EcoreValidator`, and it checks essentially for valid Java identifiers.

File: emf/src/test/resources/properties.ecore (hunk 3 above)
Comment (enikao) [reply emf/src/test/resources/properties.ecore:3]: invalid, see above

File: emf/src/main/java/io/lionweb/lioncore/java/emf/EMFMetamodelExporter.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (ftomassetti) [reply emf/src/main/java/io/lionweb/lioncore/java/emf/EMFMetamodelExporter.java:35]: Makes sense, I would just add a comment as others could be tempted to remove it when we forget the reason for this

File: emf/src/test/resources/properties.ecore (hunk 3 above)
Comment (ftomassetti) [reply emf/src/test/resources/properties.ecore:3]: Understood, thank you

**Comment by AtanasM98** [comment]
```
Caused by: org.eclipse.emf.ecore.resource.impl.ResourceSetImpl$1DiagnosticWrappedException: java.io.FileNotFoundException: 
C:\Work\lioncore-java\emf-builtins\build\resources\main\src\main\model\builtins.genmodel
(The system cannot find the path specified)
```
I'm having this problem. I think it may be connected to a missing gradle task or some task that sets up this resource isn't depended on.

**Comment by enikao** [comment]
> ```
> Caused by: org.eclipse.emf.ecore.resource.impl.ResourceSetImpl$1DiagnosticWrappedException: java.io.FileNotFoundException: 
> C:\Work\lioncore-java\emf-builtins\build\resources\main\src\main\model\builtins.genmodel
> (The system cannot find the path specified)
> ```
> 
> I'm having this problem. I think it may be connected to a missing gradle task or some task that sets up this resource isn't depended on.

That should be fixed by now, could you try?

**Pull request merged**